File: Example_VI_CtCI_Python/Example14.py
def allFib(n):
    i = 0
    while (i <= n):
        print(str(i) + ": " + str(fib(i)))
        i += 1

def fib(n):
    if (n <= 0):
        return 0
    elif (n == 1):
        return 1
    return fib(n - 1) + fib(n - 2)

n = 20
allFib(n)

File: Examples_VI_CtCI_CPP/Example 14/Example14.cpp
#include <iostream>
// O (2^N) runtime.

int fib(int n) {
	if (n <= 0) return 0;
	else if(n == 1) return 1;
	return fib(n - 1) + fib (n - 2);
}

void allFib(int n) {
	for (int i = 0; i <= n; i++) {
		std::cout << i << ": " << fib(i) << std::endl;
	}	
}

int main() {
	int n;
	allFib(20);
}
